paperwork_backend/docsearch.py

```diff
@@ -264,16 +264,21 @@ def __init__(self, rootdir, indexdir=None, language=None,
 
         self.fs = fs.GioFileSystem()
         self.rootdir = self.fs.safe(rootdir)
-
         localdir = os.path.expanduser("~/.local")
-        if indexdir is None:
+
+        if index_in_workdir == True:
+            self.rootdir = self.fs.unsafe(rootdir)
```

Comment:
One minor change and you're good to go:

With your current changes, when `index_in_workdir` is False, `self.rootdir` is an URI. When `index_in_workdir` is True, `self.rootdir` is a Unix path. Having 2 different types of values for a single attribute can be really bug-prone.
I would prefer that `self.rootdir` always remain an URI.

Here it can be simply fixed by not changing `self.rootdir` ; with something like:

```py
if index_in_workdir:
    base_data_dir = self.fs.unsafe(rootdir)
    localdir = base_data_dir
    indexdir = os.path.join(base_data_dir, "index")
```